### PR TITLE
delete useless code

### PR DESCRIPTION
**Pull request by alaric27**
These codes confuse me，I think these code is useless

### CODE REVIEW

File: broker/src/main/java/org/apache/rocketmq/broker/processor/PullMessageProcessor.java [hunk 1]
```diff
@@ -246,12 +246,6 @@ private RemotingCommand processRequest(final Channel channel, RemotingCommand re
             responseHeader.setMinOffset(getMessageResult.getMinOffset());
             responseHeader.setMaxOffset(getMessageResult.getMaxOffset());
 
-            if (getMessageResult.isSuggestPullingFromSlave()) {
```
Comment (hzh0425) [inline broker/src/main/java/org/apache/rocketmq/broker/processor/PullMessageProcessor.java:249]: why it useless?

Comment (hzh0425) [reply broker/src/main/java/org/apache/rocketmq/broker/processor/PullMessageProcessor.java:249]: This is the code used to relieve the pressure of the Master

Comment (alaric27) [reply broker/src/main/java/org/apache/rocketmq/broker/processor/PullMessageProcessor.java:249]: I think the following code covers this logic
`
            if (this.brokerController.getBrokerConfig().isSlaveReadEnable()) {
                // consume too slow ,redirect to another machine
                if (getMessageResult.isSuggestPullingFromSlave()) {
                    responseHeader.setSuggestWhichBrokerId(subscriptionGroupConfig.getWhichBrokerWhenConsumeSlowly());
                }
                // consume ok
                else {
                    responseHeader.setSuggestWhichBrokerId(subscriptionGroupConfig.getBrokerId());
                }
            } else {
                responseHeader.setSuggestWhichBrokerId(MixAll.MASTER_ID);
            }
`

Comment (hzh0425) [reply broker/src/main/java/org/apache/rocketmq/broker/processor/PullMessageProcessor.java:249]: The logic is not the same

**Comment by ni-ze** [comment]
It does look repeated.
If this.brokerController.getBrokerConfig().isSlaveReadEnable() equals false, suggest broker is always master.

```java
          if (getMessageResult.isSuggestPullingFromSlave()) {
                responseHeader.setSuggestWhichBrokerId(subscriptionGroupConfig.getWhichBrokerWhenConsumeSlowly());
            } else {
                responseHeader.setSuggestWhichBrokerId(MixAll.MASTER_ID);
            }
```
Result of the above code is covered.

If slaveReadEnable equals true, the result of latter code part has also cover the above.

**Review by tsunghanjacktsai — APPROVED** [review]
Both of them do have a duplicate effect in the end actually. I believe we could remove one of them.

File: broker/src/main/java/org/apache/rocketmq/broker/processor/PullMessageProcessor.java (hunk 1 above)
Comment (tsunghanjacktsai) [reply broker/src/main/java/org/apache/rocketmq/broker/processor/PullMessageProcessor.java:249]: @hzh0425 I believe it's fine to remove it 'cause they do have the same effect in the end, but welcome to provide your thoughts on this.

**Pull request closed without merging**